feat(environment): template publication

Davidmattei opened the pull request:
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Add the possibility to define a publication template on an environment.
If defined we should check the publication template before publication.

Created a EnvironmentPublisher service with his own factory.
The factory will remember previous publishers, for using it during a bulk publication

https://ems-project.github.io/#/./elasticms/environment/environment

Davidmattei committed:
feat(environment): add template publication

Davidmattei committed:
fix(code-editor): min lines not working

Davidmattei committed:
fix(color-picker): do not translate choices (colors)

Davidmattei committed:
feat(environment): edit template publication

add new translation domain "environment"

Davidmattei committed:
feat(environment): environment publisher

theus77 commented on src/Form/Form/EditEnvironmentType.php:
```diff
@@ -22,51 +23,62 @@ class EditEnvironmentType extends AbstractType
     public function configureOptions(OptionsResolver $resolver): void
     {
-        /* set the default option value for this kind of compound field */
-        parent::configureOptions($resolver);
```
Are you sure that it's not required to call the parent's configureOptions fucntion?

Davidmattei replied:
yes, the parent class is Symfony\Component\Form\AbstractType; which has no implementation on configureOptions.

theus77 commented on src/Service/Revision/RevisionService.php:
```diff
@@ -186,6 +186,11 @@ public function getDocumentInfo(EMSLink $documentLink): DocumentInfo
         return new DocumentInfo($documentLink, $this->revisionRepository->findAllPublishedRevision($documentLink));
     }
 
+    public function getDocumentInfoByRevision(Revision $revision): DocumentInfo
```
Does the DocumentInfo entity is abel to know if a revision can be published in a specific environment (for overview tables)?

Davidmattei replied:
Good remark this change has been removed, did not need the documentInfo but the document.

If you want the documentInfo in the publicaton template -> revison.emsLink|emsco_document_info

Davidmattei committed:
feat(environment): roll revisionService change

Davidmattei committed:
feat(environment): remove unused extra column

Davidmattei committed:
fix: revision newVersionTag not defined

Davidmattei committed:
fix(environment): extra field removed

Davidmattei committed:
fix(contentType): could not select date fields for version

Davidmattei committed:
fix(environment): align publication check and with version revision

Davidmattei committed:
feat(environment): publication cli emsco:environment:align

Davidmattei committed:
chore: moved documentation

https://github.com/ems-project/ems-project.github.io/commit/1b107781137de788ebd35c367c496c7aa51eb86b
